src/components/vuestic-components/va-form/VaForm.vue

```diff
@@ -0,0 +1,162 @@
+<script>
+import { ContextPluginMixin, getContextPropValue } from '../../context-test/context-provide/ContextPlugin'
+
+const availableFormTags = ['VaInput', 'VaColorInput', 'VaDatePicker', 'VaCheckbox', 'VaRadio', 'VaSelect', 'VaFileUpload']
+const getAllChildren = (vm, children = []) => {
+  vm.$children.forEach(function (child) {
+    children.push(child)
+    child.$children.length > 0 && getAllChildren(child, children)
+  })
+  return children
+}
+
+export default {
+  name: 'VaForm',
+  mixins: [ContextPluginMixin],
+  props: {
+    lazyValidation: {
+      type: Boolean,
+      default () {
+        return getContextPropValue(this, 'lazyValidation', false)
+      },
+    },
+    autofocus: {
+      type: Boolean,
+      default () {
+        return getContextPropValue(this, 'autofocus', false)
+      },
+    },
+    tag: {
+      type: String,
+      default () {
+        return getContextPropValue(this, 'tag', 'div')
+      },
+    },
+  },
+  data () {
+    return {
+      valid: true,
+      hasResetButton: false,
+      hasSubmitButton: false,
+    }
+  },
+  mounted () {
+    this.$el.addEventListener('focusin', this.focus)
+    const formChild = getAllChildren(this)
+
+    if (this.autofocus) {
+      const firstFormChild = formChild.find((child) => availableFormTags.includes(child.$options.name))
+
+      if (firstFormChild) {
+        firstFormChild.$el.focus()
+      }
+    }
+
+    if (this.tag !== 'form') { // component dont have tag form, we need set listeners manually
+      const resetButton = this.$el.querySelector('button[type=reset]')
+
+      if (resetButton) {
+        resetButton.addEventListener('click', this.reset)
+
+        this.hasResetButton = true
+      }
+
+      const submitButton = this.$el.querySelector('button[type=submit]')
+
+      if (submitButton) {
+        submitButton.addEventListener('click', this.submit)
+
+        this.hasSubmitButton = true
+      }
+    }
+  },
+  beforeDestroy () {
+    this.$el.removeEventListener('focusin', this.focus)
+
+    if (this.hasSubmitButton) {
```

Comment:
Pretty sure that's pointless. Are you worried vue will reuse the same dom element?

Reply:
In some browsers memory leak is a big problem.
https://stackoverflow.com/questions/6033821/do-i-need-to-remove-event-listeners-before-removing-elements

Reply:
In issue you sent it says we're fine for IE11+ which is the case for vuestic.
So please remove if you don't have any other reasons to keep this code.

Reply:
Fixed